src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -632,8 +638,100 @@ private static ITaskSeriesTimer StartParameterLogTimer(IRecurrentCommand updateC
+                throw;
+            }
+            finally
+            {
+                await InvokeExecutedFilters(filters, parameterNames, invokeParameters, functionCancellationTokenSource, instance, logger, jobHost, config, functionResult);
+            }
+        }
+
+        private static List<IFunctionInvocationFilter> GetFilters(IFunctionInstance instance, IFunctionInvoker invoker)
+        {
+            List<IFunctionInvocationFilter> filters = new List<IFunctionInvocationFilter>();
+
+            if (invoker.Instance != null)
+            {
+                var instanceFilter = invoker.Instance as IFunctionInvocationFilter;
+
+                if (instanceFilter != null)
+                {
+                    filters.Add(instanceFilter);
+                }
+            }
+
+            var methodFilters = instance.FunctionDescriptor.Method.GetCustomAttributes().OfType<IFunctionInvocationFilter>();
+            filters.AddRange(methodFilters);
+
+            var classFilters = instance.FunctionDescriptor.Method.DeclaringType.GetCustomAttributes().OfType<IFunctionInvocationFilter>();
+            filters.AddRange(classFilters);
+
+            return filters;
+        }
 
-            await invokeTask;
+        private static async Task InvokeExecutingFilters(List<IFunctionInvocationFilter> filters, IReadOnlyList<string> invokeParameterNames, object[] invokeParameters, CancellationTokenSource functionCancellationTokenSource,
+            IFunctionInstance instance, ILogger logger, JobHost jobHost, JobHostConfiguration config)
+        {
+            if (filters.Any())
+            {
+                // Setup the parameters
+                var parameters = new Dictionary<string, object>();
+
+                for (var i = 0; i < invokeParameters.Length; i++)
+                {
+                    parameters.Add(invokeParameterNames[i], invokeParameters[i]);
+                }
+
+                // Create the context
+                FunctionExecutingContext context = new FunctionExecutingContext(instance.Id, instance.FunctionDescriptor.FullName, parameters, logger);
+                context.JobHost = jobHost;
+                context.Config = config;
+
+                foreach (var filter in filters)
+                {
+                    await filter.OnExecutingAsync(context, functionCancellationTokenSource.Token);
+                }
+            }
+        }
+
+        private static async Task InvokeExecutedFilters(List<IFunctionInvocationFilter> filters, IReadOnlyList<string> invokeParameterNames, object[] invokeParameters, CancellationTokenSource functionCancellationTokenSource,
+            IFunctionInstance instance, ILogger logger, JobHost jobHost, JobHostConfiguration config, FunctionResult functionResult)
+        {
+            if (filters.Any())
+            {
+                // Setup the parameters
+                var parameters = new Dictionary<string, object>();
```

Comment:
It's all duplicated code between these two functions. We can compress into a single function, since the only thing that varies between the two is which method is called (Executing/Executed). You can just have a single InvokeFiltersAsync method that takes a Func<IFunctionInvocationFilter, Task> param to do the invoke.